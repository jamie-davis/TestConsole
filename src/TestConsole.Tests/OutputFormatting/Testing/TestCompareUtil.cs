using System.Linq;
using NUnit.Framework;
using TestConsoleLib;
using TestConsoleLib.Testing;

namespace TestConsole.Tests.OutputFormatting.Testing
{
    [TestFixture]
    public class TestCompareUtil
    {
        [ApprovalsReporter("testtool")]
        public class TestToolReporter : IApprovalsReporter
        {
            public string FileName => "testtool";
            public string Arguments => "$1 $2";
        }

        [Test]
        public void ReportersCanBeAdded()
        {
            //Arrange
            CompareUtil.RegisterReporters(GetType().Assembly);
            
            //Act/Assert
            var output = new Output();
            output.FormatTable(CompareUtil.ListReporters().Select(t => new {t.ReporterName, t.ReporterType}));
            output.Report.Verify();
        }

        [Test]
        public void ReportersCanBeAddedMultipleTimesWithoutProblems()
        {
            //Act
            CompareUtil.RegisterReporters(GetType().Assembly);
            CompareUtil.RegisterReporters(GetType().Assembly);
            CompareUtil.RegisterReporters(GetType().Assembly);
            CompareUtil.RegisterReporters(GetType().Assembly);
            CompareUtil.RegisterReporters(GetType().Assembly);

            //Assert
            var output = new Output();
            output.FormatTable(CompareUtil.ListReporters().Select(t => new {t.ReporterName, t.ReporterType}));
            output.Report.Verify();
        }
    }
}
