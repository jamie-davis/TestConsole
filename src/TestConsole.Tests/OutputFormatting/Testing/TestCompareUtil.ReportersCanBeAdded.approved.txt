Reporter                                                                                       
Name     Reporter Type                                                                         
-------- ---------------------------------------------------------------------------           
testtool TestConsole.Tests.OutputFormatting.Testing.TestCompareUtil+TestToolReporter           
kompare  TestConsoleLib.Testing.KompareReporter                                                
meld     TestConsoleLib.Testing.MeldReporter                                                   
winmerge TestConsoleLib.Testing.WinMergeReporter                                               
                                                                                               